### PR TITLE
Feature/fixdeprecatedverbsinlocaltstorage

### PR DESCRIPTION
**Pull request by manudous**
i have problems with type this function but if i type to any works but i dont want this :). @brauliodiez 
image

### CODE REVIEW

**Comment by manudous** [comment]
Resolve the type, i have learned a new form to type XD
image

File: src/core/verbs/gobal-verbs-sanitizer.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const sanitizeVerbSelectionList = (
+  ...selectedList: string[][]
```
Comment (brauliodiez) [inline src/core/verbs/gobal-verbs-sanitizer.ts:2]: Esto no lo entiendo, si tienes dos parametros, porque no poner dos parametros?

Igual puse yo mal la firma

File: src/core/verbs/gobal-verbs-sanitizer.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+): string[] =>
+  selectedList.reduce((acc, item) => acc.filter(x => item.indexOf(x) > -1)) ?? [];
```
Comment (brauliodiez) [inline src/core/verbs/gobal-verbs-sanitizer.ts:4]: Aqui partimos de que tenemos dos variables:
- Una es la lista de verbos seleccionados que hemos leído de local storage, que es la que puede tener alguna entrada que esté mal.
- Otra la lista total de verbos, que eso lo podemos pasar por parametro o directamente consumirlo.

Podíamos tener algo así (en el home.container):

```ts
    selectedVerbs = sanitizeVerbSelectionList(
      selectedVerbs,
     fullVerbCollection
    );
```

El sanitize tendría esta firma

```ts
export const sanitizeVerbSelectionList(selection: string[], fullVerbCollection: VerbEntityGlobal[])

```

File: src/core/verbs/gobal-verbs-sanitizer.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const sanitizeVerbSelectionList = (a: string[], b: string[]): string[] =>
```
Comment (brauliodiez) [inline src/core/verbs/gobal-verbs-sanitizer.ts:1]: Do not use "a" and "b", they have no domain meaning

We could use:

currentVerbSelection, fullVerbList

Usually avoid having variable names like "a", "b", "i"... unless you are using them in a very limites scope (for instance inside a filter)

File: src/core/verbs/global-verbs-sanitizer.spec.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    // Act
+    const result: string[] = sanitizeVerbSelectionList(
+      verbListOne,
+      verbListTwo
+    );
+
+    // Assert
+    expect(result).toEqual([]);
+  });
+  it('should return one item when it feeds sanitizeVerbSelectionList with two items', () => {
```
Comment (brauliodiez) [inline src/core/verbs/global-verbs-sanitizer.spec.ts:46]: There are more cases missing, e.g.:

1. None of the verbs are in the list should return empty array
2. Some of the verbs are in the list but not the first (ej 3) ones in the array.
3. Some of the verbs are in the list but the last (ej 3) in the array
4. Some of the verbs are in the list but they are mixed in the array (Some of the are some of them not)

**Pull request merged**